WordPress/src/main/java/org/wordpress/android/ui/mlp/LayoutsAdapter.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.mlp
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.ImageView.ScaleType.FIT_CENTER
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.modal_layout_picker_layouts_card.view.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageType
+import org.wordpress.android.util.setVisible
+import javax.inject.Inject
+
+/**
+ * Renders the Layout cards
+ */
+class LayoutsAdapter(
+    private val context: Context,
+    private val selectionListener: LayoutSelectionListener
+) : RecyclerView.Adapter<LayoutsAdapter.ViewHolder>() {
+    @Inject lateinit var imageManager: ImageManager
+
+    private var layouts: List<LayoutListItem> = listOf()
+
+    init {
+        (context.applicationContext as WordPress).component().inject(this)
+    }
+
+    fun setData(data: List<LayoutListItem>) {
+        val diffResult = DiffUtil.calculateDiff(
+                LayoutsDiffCallback(
+                        layouts,
+                        data
+                )
+        )
+        layouts = data
+        diffResult.dispatchUpdatesTo(this)
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): ViewHolder {
+        val v = LayoutInflater.from(parent.context)
+                .inflate(R.layout.modal_layout_picker_layouts_card, parent, false)
+        return ViewHolder(v)
+    }
+
+    override fun getItemCount(): Int {
+        return layouts.size
+    }
+
+    override fun onBindViewHolder(
+        holder: ViewHolder,
+        position: Int
+    ) {
+        val layout = layouts[position]
+        imageManager.load(holder.preview, ImageType.THEME, layout.preview, FIT_CENTER)
+        holder.selected.setVisible(layout.selected)
+        holder.preview.contentDescription = if (layout.selected) context.getString(
+                R.string.mlp_layout_selected,
+                layout.title
+        ) else layout.title
+        holder.container.setOnClickListener {
+            selectionListener.layoutTapped(layout)
+        }
+        selectionListener.selectedItemData.observe(selectionListener.lifecycleOwner, Observer {
```

Comment:
Imo viewHolder shouldn't be responsible for listening to state changes - ideally, it'd just receive data and render them. When something needs to be changed the vm updates the data - passes them to the adapter - the viewholder re-renders them. Wdyt?

Reply:
I agree. Fixed with 05962fa0f4e6eff4c9a13e5372348929e172f06f